docs: Node.js runtime support for Middleware (#76556)
vercel/next.js: docs/01-app/03-building-your-application/01-routing/14-middleware.mdx

> Middleware currently only supports APIs compatible with the [Edge runtime](/docs/app/building-your-application/rendering/edge-and-nodejs-runtimes). APIs exclusive to Node.js are [unsupported](/docs/app/api-reference/edge#unsupported-apis).
> 
> ### Using Node.js runtime with middleware
> 
> The Node.js runtime can be used as an experimental feature to run middleware. To enable, add the flag to your `next.config.js` file:

Reviewer: ```suggestion
Middleware defaults to using the Edge runtime. As of v15.2, we have experimental support for using the Node.js runtime. To enable, add the flag to your `next.config` file:
```

```ts filename="next.config.ts" switcher
import type { NextConfig } from 'next';

const nextConfig: NextConfig = {
  experimental: {
    nodeMiddleware: true,
  },
};

export default nextConfig;
```

```js filename="next.config.js" switcher
const nextConfig = {
  experimental: {
    nodeMiddleware: true,
  },
};

export default nextConfig;
```

Then in your middleware file, set the runtime to `nodejs` in the `config` object:

```js highlight={2} filename="middleware.js" switcher
export const config = {
  runtime: 'nodejs',
};
```

```ts highlight={2} filename="middleware.ts" switcher
export const config = {
  runtime: 'nodejs',
};
```

## Version History

| Version   | Changes                                                                                       |
| --------- | --------------------------------------------------------------------------------------------- |

> | `v15.2.0` | Middleware can now use the Node.js runtime. See the [middleware documentation](/docs/app/building-your-application/routing/middleware#using-nodejs-runtime-with-middleware) for more details. |

Reviewer: ```suggestion
| `v15.2.0` | Middleware can now use the Node.js runtime (experimental) |
```
